profiles/drupalru/modules/dru_tnx/dru_tnx.install

```diff
@@ -4,135 +4,271 @@
-  db_add_field('node', 'tnx', array(
+  db_add_field('node', 'tnx', [
     'type'        => 'int',
     'not null'    => TRUE,
     'default'     => 0,
     'description' => 'Thanks count.',
-  )); 
-  db_add_index('node', 'tnx', array('tnx'));
-  
+  ]);
+  db_add_index('node', 'tnx', ['tnx']);
+
   // Add tnx field to comment.
-  db_add_field('comment', 'tnx', array(
+  db_add_field('comment', 'tnx', [
     'type'        => 'int',
     'not null'    => TRUE,
     'default'     => 0,
     'description' => 'Thanks count.',
-  )); 
-  db_add_index('comment', 'tnx', array('tnx'));
-  
+  ]);
+  db_add_index('comment', 'tnx', ['tnx']);
+
 }
 
 /**
  * Implements hook_schema_alter().
  */
 function dru_tnx_schema_alter(&$schema) {
-    
+
   // Add tnx field to node.
-  $schema['node']['fields']['tnx'] = array(
+  $schema['node']['fields']['tnx'] = [
     'description' => 'Thanks count.',
     'type'        => 'int',
     'not null'    => TRUE,
     'default'     => 0,
-  );
-  
+  ];
+
   // Add tnx field to comment.
-  $schema['comment']['fields']['tnx'] = array(
+  $schema['comment']['fields']['tnx'] = [
     'description' => 'Thanks count.',
     'type'        => 'int',
     'not null'    => TRUE,
     'default'     => 0,
-  );
+  ];
 
 }
 
 /**
+ * Recount thanks statistic.
+ */
+function dru_tnx_update_7000(&$sandbox) {
+
+  $data = [];
+  $users = [];
+  $delete = [];
+  $nodes = [];
+  $comments = [];
+
+  if (!isset($sandbox['total'])) {
+    $query = db_select('tnx', 't')
+      ->fields('t', [])
+      ->execute();
+
+    $results = $query->fetchAll();
+
+    foreach ($results as $item) {
+
+      if (isset($data[$item->entity_type][$item->entity_id][$item->uid])) {
+        $delete[] = $item->id;
+        continue;
+      }
+
+      $data[$item->entity_type][$item->entity_id][$item->uid] = $item->id;
+      $users[$item->uid]['tnx_node'] = 0;
+      $users[$item->uid]['tnx_comment'] = 0;
+      $users[$item->uid][$item->entity_type][$item->entity_id] = $item->id;
+    }
+
+    if (count($delete)) {
+      try {
+        db_query("DELETE FROM tnx WHERE id IN (" . implode(",", $delete) . ")");
+      } catch (\Exception $exception) {
+        throw new \Exception('Can\'t delete doubles!');
+      }
+    }
+
+    if (count($data['node'])) {
+      try {
+        $nodes = db_query(
+          "SELECT nid, uid FROM node WHERE nid IN (" . implode(",",
+            array_keys($data['node'])) . ")"
+        )->fetchAllKeyed();
+      } catch (\Exception $exception) {
+        throw new \Exception('Can\'t load all nodes!');
+      }
+    }
+
+    if (count($data['comment'])) {
+      try {
+        $comments = db_query("SELECT cid, uid FROM comment WHERE cid IN (" . implode(",",
+            array_keys($data['comment'])) . ")"
+        )->fetchAllKeyed();
+      } catch (\Exception $exception) {
+        throw new \Exception('Can\'t load all nodes!');
+      }
+    }
+
+    if ($nodes) {
+      foreach ($nodes as $id => $uid) {
```

Comment:
This is pretty hard code! I think we will have troubles when deployment will started.